Merge branch-SortEvents with Master (#113)
AY2223S1-CS2103-F09-2/tp: docs/UserGuide.md

* If a parameter is expected only once in the command, but you specified it multiple times, only the last occurrence of the parameter will be taken.<br>
  e.g. if you specify `p/12341234 p/56785678`, only `p/56785678` will be taken.

> * Extraneous parameters for commands that do not take in parameters (such as `help`, `listPersons`, `exit` and `clear`) will be ignored.<br>

Reviewer: Our listPersons does takes in an optional parameter for sorting so it should not fall under this category of commands that do not take in parameters. Shall we remove listPersons from this?

- Adds date of birth to a person in the contact list. Date format accepted is: dd/mm/yyyy.

### Listing all persons : `listPersons`

Shows a list of all persons in the application.

Format: `listPersons [s/FIELD]`

* Sorts the contacts by the specified field in **ascending** order. `FIELD` must take one the following values:
  * `n` or `N` sort by name ignoring case differences
  * `d` or `D` sort by date of birth
  * `g` or `G` sort by gender

> * It is optional to specify the field to sort by. If no field is specified, no sorting is performed.

Reviewer: Shall we change it to this? 
```suggestion
* It is optional to include the sorting prefix and field. If the sorting prefix and field to sort by are not included, no sorting is performed.
```
Possible that this 'If no field is specified' might be interpreted as having "listPersons s/" without the field

> * Persons with an empty field that is being used to sort will be placed at the top of the list.

Reviewer 2: Maybe this line can be removed since gender, date of birth and name are all compulsory fields for a person now.

<div markdown="span" class="alert alert-info">

> :information_source: **Note:** The sorted result is permanent on the underlying contacts list.<br><br>

Reviewer: Just to standardise with the rest!
```suggestion
:information_source: **Note:** The sorted result is permanent on the underlying contact list.<br><br>
```

Reviewer: On a side note, some other members have referred to this list as person list, so we might need to change this again depending on the convention we decide on

* `deleteEvent 2` after listing all events with `listEvents` deletes the event at index 2
* `deleteEvent 7` after listing all events with `listEvents` deletes the event at index 7

### Listing all events: `listEvents`

Shows a list of all events in the application.

Format: `listEvents [s/FIELD]`

* Sorts the contacts by the specified field in **ascending** order. `FIELD` must take one the following values:

>   * `n` or `N` sort by name ignoring case differences

Reviewer: We'll be using the prefix "e" to denote the event's title (i.e eventTitle attribute of the Event class) 

```suggestion
* Sorts the events by the specified field in **ascending** order. `FIELD` must take one the following values:
  * `e` or `E` sort by event title ignoring case differences
```

  * `d` or `D` sort by date

* It is optional to specify the field to sort by. If no field is specified, no sorting is performed.

> * At most one field can be specified. i.e. Cannot specify 2nd or 3rd criteria to sort by.

Reviewer: Same suggestions as before, if you're modifying the above, pls edit this too!

<div markdown="span" class="alert alert-info">
:information_source: **Note:** The sorted result is permanent on the underlying events list.<br><br>

> For example, if `listEvents s/n` and `listEvents` are executed back-to-back, the result of the second `listEvents` command will display the sorted results from the first `listEvents s/n` command since the sorted result is permanent.

Reviewer: Minor change to 'e' for eventTitle
```suggestion
For example, if `listEvents s/e` and `listEvents` are executed back-to-back, the result of the second `listEvents` command will display the sorted results from the first `listEvents s/e` command since the sorted result is permanent.
```

</div>

Examples:
* `listEvents` Lists all events without performing any sorting.

> * `listEvents s/n` Lists all events sorted by their names.

Reviewer: Minor change to 'e' for event title
```suggestion
* `listEvents s/e` Lists all events sorted by their event titles.
```

| **Delete**      | `delete INDEX`<br> e.g., `delete 3`                                                                                                                                                    |
| **Edit**        | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [g/GENDER] [d/DOB] [t/TAG]`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`                                                  |
| **Find**        | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`                                                                                                                             |

> | **ListPersons**        | `listPersons  [s/FIELD]` <br> e.g., `listPersons s/n`                                                                                                                                                |

Reviewer 2: ```suggestion
| **ListPersons**        | `listPersons [s/FIELD]` <br> e.g., `listPersons s/n`                                                                                                                                                |
```

| **AddEvent**    | `addEvent n/EVENT_TITLE d/DATE t/TIME p/PURPOSE`<br> e.g.,`addEvent n/Shoe Sale 30% d/30-05-2022 t/11:00 p/Discount on all shoes for up to 30%`                                        |
| **DeleteEvent** | `deleteEvent INDEX`<br> e.g., `deleteEvent 2`                                                                                                                                          |

> | **ListEvents**  | `listEvents [s/FIELD]`<br> e.g., `listEvents s/n`                                                                                                                                                                           |

Reviewer: Minor change to 'e' for event title
```suggestion
| **ListEvents**  | `listEvents [s/FIELD]`<br> e.g., `listEvents s/e`                                                                                                                                                                           |
```
